### ISSUE TITLE
Proposal: Control of the operating system through BabyAGI

### ISSUE THREAD

**Issue by Radhe-0** [yoheinakajima/babyagi#74]
I have a pretty good idea that may become key to an artificial intelligence capable of performing tasks on a computer (at least based on Linux). The idea that occurs to me is very aligned with this project, it is the following:

1. The task list will be created and distributed to other sub agents that will be in charge of creating various processes including creating .sh files (to execute linux commands) and visual feedback for GPT-4. When this feedback occurs, another agent will create a .sh file that will continue processing until the assigned goal is met.

2. This will give babyAGI the ability to start accomplishing goals, but within the operating system, from small goals like "Open a program" to more complex goals like getting things done within the internet browser in conjunction with other programs.

3. This would be relatively easy to test, since it doesn't need to "execute" all those instructions, just that the output is the list of commands in order to be executed, so there is little risk.

4. It is also possible that the commands to be executed include mouse and keyboard events. Also, screenshots that serve as feedback for BabyAGI to continue interacting with the computer.

In short, given a goal as input, BabyAGI creates the commands for the Linux distribution to meet that goal, the output would be the interaction with the computer. It would be a great experiment because it aims to give your operating system "own brain and decision power".

The implementation of this proposal could be considered an important step towards the development of a strong artificial intelligence, since it would allow the AI to interact with the physical world through a computer.

**Comment by micromysore** [comment]
something like this ? https://github.com/zahidkhawaja/rusty

**Comment by Radhe-0** [comment]
> something like this ? https://github.com/zahidkhawaja/rusty

It is a good start

**Comment by PrudhviChakravarthy** [comment]
i have similar idea, where users do some commands of some task on Linux like - installing YT-dlp and downloading audio or video where ai just create the whole process using bash commands and push it to some private repository. if any one ask the bot to download same using prompt it checks the existing codes in it repository and download necessary bash scripts and run it. 

Make sure it checks the script before running for malfunctions and bad practices.

**Comment by francip** [comment]
Check out https://github.com/yoheinakajima/babyagi/pull/85 - this allows us to actually build "companions" that can do specific things like what you propose. The main BabyAGI then would create tasks which are annotated with [os] for example, and the companion will pick those and execute.
